docs/_docs/reference/contextual/context-bounds.md

```diff
@@ -4,50 +4,205 @@ title: "Context Bounds"
 nightlyOf: https://docs.scala-lang.org/scala3/reference/contextual/context-bounds.html
 ---
 
-A context bound is a shorthand for expressing the common pattern of a context parameter that depends on a type parameter. Using a context bound, the `maximum` function of the last section can be written like this:
+A context bound is a shorthand for expressing the common pattern of a context parameter that depends on a type parameter. These patterns are commonplace when modelling type classes in Scala. Using a context bound, the `maximum` function of the [last section](./using-clauses.md) can be written like this:
 
 ```scala
 def maximum[T: Ord](xs: List[T]): T = xs.reduceLeft(max)
 ```
 
-A bound like `: Ord` on a type parameter `T` of a method or class indicates a context parameter `using Ord[T]`. The context parameter(s) generated from context bounds
-are added as follows:
+A bound like `: Ord` on a type parameter `T` of a method or class indicates a context parameter `using Ord[T]`, which is added to the signature of the enclosing method. The generated parameter is called a _witness_ for the context bound.
 
- - If the method parameters end in an implicit parameter list or using clause,
-   context parameters are added in front of that list.
- - Otherwise they are added as a separate parameter clause at the end.
-
-Example:
+For instance the `maximum` method above expands to
+```scala
+def maximum[T](xs: List[T])(using Ord[T]): T = ...
+```
+Context bounds can be combined with subtype bounds. If both are present, subtype bounds come first, e.g.
 
 ```scala
-def f[T: C1 : C2, U: C3](x: T)(using y: U, z: V): R
+def f[T <: B : C](x: T): R = ...
 ```
 
-would expand to
+## Named Context Bounds
 
+A context bound can be given a name with an `as` clause. For example, assume the following trait definitions.
+```scala
+  trait SemiGroup[A]:
+    extension (x: A) def combine(y: A): A
+
+  trait Monoid[A] extends SemiGroup[A]:
+    def unit: A
+```
+We can write `reduce` function over lists of monoid instances like this:
 ```scala
-def f[T, U](x: T)(using _: C1[T], _: C2[T], _: C3[U], y: U, z: V): R
+  def reduce[A: Monoid as m](xs: List[A]): A =
+    xs.foldLeft(m.unit)(_ `combine` _)
 ```
+We use `as x` after the type of a context bound to bind the instance to `x`. This is analogous to import renaming, which also introduces a new name for something that comes before.
```

Comment:
Perhaps there should be an example showing where the `as` clauses are attached when there are more than one context bound.

Reply:
We have that a little bit later.